app/models/topic_link.rb

```diff
@@ -289,6 +289,10 @@ def self.ensure_entry_for(post, link, parsed)
       # Store the same URL that will be used in the cooked version of the post
       url = UrlHelper.cook_url(upload.url, secure: upload.secure?)
     elsif route = Discourse.route_for(parsed)
+      # this is a special case for the silent flag
+      # in internal links
+      return nil if url && (url.split("?")[1] == "silent=true")
```

Comment:
I think we may have to do this:

```ruby
CGI.parse(UrlHelper.relaxed_parse("http://localhost:4200/t/1234?blah=true&silent=true").query)
=> {"blah"=>["true"], "silent"=>["true"]}
```

Otherwise we miss the edge case of `silent=true` being used after `&`. It's odd that we don't have a `UrlHelper` method already to extract query params like this. I guess alternatively we can just do:

```ruby
return nil if url&.include?("silent=true")
```

Reply:
not sure I want to technically support this ... the main use case for needing it in & would be that it is a user link. 

And then silent probably should be overridden: 

Eg: <https://meta.discourse.org/t/daily-summary-feedback/291853/99?u=sam&silent=true> 

This is not good (tm) cause I am trying to track it as my link AND get away with it being silent. I prefer to be atomic here for starters.

Reply:
No probs :+1: